lib/Backend/GlobOpt.cpp

```diff
@@ -6935,20 +6960,34 @@ GlobOpt::SetSymStore(ValueInfo *valueInfo, Sym *sym)
     }
     if (valueInfo->GetSymStore() == nullptr || valueInfo->GetSymStore()->IsPropertySym())
     {
-        valueInfo->SetSymStore(sym);
+        SetSymStoreDirect(valueInfo, sym);
     }
 
     return sym;
 }
 
 void
+GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
+{
+    Sym * prevSymStore = valueInfo->GetSymStore();
+    if (prevSymStore && prevSymStore->IsStackSym() &&
+        prevSymStore->AsStackSym()->HasByteCodeRegSlot() &&
+        this->blockData.changedSyms)
```

Comment:
I don't see a path where changedSyms would be NULL.  You should remove the NULL checks.

---
In reply to: [68132431](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68132431) [](ancestors = 68132431,68113857,68113107)